File: 0027 - Remove Element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i, k = 0, 0
        while i < len(nums):
            if nums[i] == val:
                k += 1
            else:
                nums[i - k] = nums[i]
            i += 1
        return i - k

File: 0027 - Remove Element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int i = 0;
        int k = 0;
        while(i < nums.size()){
            if(nums[i] == val){k++;}
            else{nums[i - k] = nums[i];}
            i++;
        }
        return i - k; 
    }
};